fem/fe/fe_base.cpp

```diff
@@ -1044,13 +1085,49 @@ void VectorFiniteElement::SetDerivMembers()
                break;
             case 1:
                deriv_type = NONE;
-               deriv_range_type = SCALAR;
-               deriv_map_type = INTEGRAL;
+               deriv_range_type = UNKNOWN_RANGE_TYPE;
+               deriv_map_type = UNKNOWN_MAP_TYPE;
```

Comment:
Why this? You can use H1 elements essentially. (That integral mapping is meaningless for points, but does not hurt for consistency)

Reply:
Hmm, I see it is not implemented in `ND_FECollection`, but that is rather missing implementation than a rule for the finite element 🤔 .

Reply:
This element does not have associated derivative space or element -- in this regard it is similar to the integral L2 element.

Reply:
There could be a special version of `PointFiniteElement` like there is `ND_R1D_PointElement`. It just needs `ProjectCurl()` or rather `CalcCurlShape()` for segments. In my code, I have used H1 space (the tangential continuity translates to pointwise continuity in 1D) for the result of curl of ND segments, but it would be better to support that natively. That "integrality" is for the 0D faces, so it is meaningless, but does not hurt 🙂 .

Reply:
This code here is for the standard `H_CURL` map type. Maybe you are thinking of the new map type `H_CURL_R1D`?

Reply:
No, I mean the sequence is complete for R1D, but for standard ND, it is incomplete for no good reason. The reviewed change just updates the element to comply with `ND_FECollection` by undefining the derivative, but we should go rather in the opposite direction perhaps, fixing the collection to follow the original definition here.

Reply:
It sounds like you are just confused -- `ND_SegmentElement` has no associated derivative and it does not have dofs at the endpoints of the segment.

Reply:
Ok, let me try to explain.

You can differentiate the basis functions of `ND_SegmentElement` in reference space, however mapping these derivatives to physical space requires derivatives of the Jacobian since `ND_SegmentElement` is mapped to physical space using the `H_CURL` map type:

 $$\hat{u}(\hat{x}) \to u(x) = J \left(J^t J\right)^{-1} \hat{u}(\hat{x})$$

where $J$ can be $1\times 1$, $2\times 1$, or $3\times 1$. This type of mapping of derivatives is not something we support. Note that for $J$ that is $1\times 1$ the `H_CURL` map type becomes the `INTEGRAL` map type:

$$\hat{u}(\hat{x}) \to u(x) = \left(1/\det(J)\right) \hat{u}(\hat{x})$$

which is another map type we do not define derivatives for.

Reply:
Ok, well there could be elements similar to `ND_R1D_SegmentElement`, but with less components and the curl shape could be implemented similarly without any derivatives of Jacobian, though it is limited to non-embedded spaces only. Anyway, I see it would not be consistent with the current definitions and defining something special for it goes beyond this PR.